packages/@aws-cdk/aws-servicecatalogappregistry/lib/application.ts

```diff
@@ -42,6 +56,45 @@ export interface ApplicationProps {
 abstract class ApplicationBase extends cdk.Resource implements IApplication {
   public abstract readonly applicationArn: string;
   public abstract readonly applicationId: string;
+  private readonly associatedAttributeGroups: Set<string> = new Set();
+  private readonly associatedResources: Set<string> = new Set();
+
+  /**
+   * Associate an attribute group with application
+   * If the attribute group is already associated, it will ignore duplicate request.
+   */
+  public associateAttributeGroup(attributeGroup: IAttributeGroup): void {
+    if (!this.associatedAttributeGroups.has(attributeGroup.node.addr)) {
+      const hashId = this.generateUniqueHash(attributeGroup.node.addr);
+      new CfnAttributeGroupAssociation(this, `AttributeGroupAssociation${hashId}`, {
+        application: this.applicationId,
+        attributeGroup: attributeGroup.attributeGroupId,
```

Comment:
Wondering - do these have to be IDs? Can these be ARNs, for cross-account capabilities, for example?

Reply:
Nope, the construct takes the ID or the name, here we are using the ID because the name field has been undergoing changes.